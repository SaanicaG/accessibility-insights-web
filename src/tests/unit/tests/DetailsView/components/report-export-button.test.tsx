// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { IButton } from '@fluentui/react';

import { render } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import { InsightsCommandButton } from 'common/components/controls/insights-command-button';
import {
    ReportExportButton,
    ReportExportButtonProps,
} from 'DetailsView/components/report-export-button';
import * as React from 'react';
import {
    mockReactComponents,
    useOriginalReactElements,
} from '../../../mock-helpers/mock-module-helpers';

jest.mock('common/components/controls/insights-command-button');

describe(ReportExportButton.displayName, () => {
    mockReactComponents([InsightsCommandButton]);
    let showDialogMock: jest.Mock<() => void> = jest.fn();
    let props: ReportExportButtonProps;

    beforeEach(() => {
        showDialogMock = jest.fn();
        props = {
            showReportExportDialog: showDialogMock,
            buttonRef: {} as React.RefObject<IButton>,
        };
    });

    it('renders ReportExportButton', () => {
        const renderResult = render(<ReportExportButton {...props} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it('shows export dialog on click', async () => {
        useOriginalReactElements('common/components/controls/insights-command-button', [
            'InsightsCommandButton',
        ]);

        const renderResult = render(<ReportExportButton {...props} />);
        await userEvent.click(renderResult.getByRole('button'));

        expect(showDialogMock).toHaveBeenCalledTimes(1);
    });
});
