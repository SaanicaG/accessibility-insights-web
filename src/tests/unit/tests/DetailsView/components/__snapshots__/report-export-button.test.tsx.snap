// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportExportButton renders ReportExportButton 1`] = `
<DocumentFragment>
  <mock-insightscommandbutton
    componentref="[object Object]"
    data-automation-id="report-export-button"
    iconprops="[object Object]"
  >
    Export result
  </mock-insightscommandbutton>
</DocumentFragment>
`;
